### PR TITLE
[MM-36789] Thread Root Post

### PR DESCRIPTION
**Pull request by AshishDhama**
#### Summary
Thread Root Post

#### Ticket Link
fixes [https://mattermost.atlassian.net/browse/MM-36789](https://mattermost.atlassian.net/browse/MM-36789)

#### Related Pull Requests
NONE

#### Screenshots
Screenshot 2021-08-25 at 6 38 59 PM

#### Release Note
```release-note
* Show ('message deleted') in global thread on root post message delete
```

### CODE REVIEW

File: components/threading/global_threads/thread_item/thread_item.tsx [hunk 1]
```diff
@@ -65,6 +66,9 @@ function ThreadItem({
 }: Props & OwnProps): React.ReactElement|null {
     const dispatch = useDispatch();
     const {select, goToInChannel} = useThreadRouting();
+    const {formatMessage} = useIntl();
+
+    const renderMessage = formatMessage({id: 'post_body.deleted', defaultMessage: '(message deleted)'});
```
[the remark this answers is not in the document]
Comment (AshishDhama) [reply components/threading/global_threads/thread_item/thread_item.tsx:71]: Yeah make sense, Done

File: components/threading/global_threads/thread_item/thread_item.tsx [hunk 2]
```diff
@@ -184,7 +188,7 @@ function ThreadItem({
                 onClick={handleFormattedTextClick}
             >
                 <Markdown
-                    message={post?.message ?? '(message deleted)'}
+                    message={post && post.state === Posts.POST_DELETED ? renderMessage : post.message}
```
[the remark this answers is not in the document]
Comment (AshishDhama) [reply components/threading/global_threads/thread_item/thread_item.tsx:191]: Thanks

**Review by koox00 — APPROVED** [review]

**Comment by stylianosrigas** [comment]
FYI the cloud server is crashing with the following error
```
{"timestamp":"2021-09-03 14:01:47.749 Z","level":"info","msg":"\u001b[36minfo\u001b[0m  [2021-09-03 14:01:47.749 Z] Server.Start                             \n\u001b[31merror\u001b[0m [2021-09-03 14:01:47.749 Z] server not bind to any port              \
```

**Review by srkgupta — APPROVED** [review]
Thanks @AshishDhama. The changes LGTM and is working fine. Just found one specific usecase which is causing the client to crash which I had Dm'd you. PTAL.

**Review by srkgupta — CHANGES_REQUESTED** [review]
Oops. Approved by mistake previously. Please see my comments above:

> Thanks @AshishDhama. Overall the changes LGTM and is working fine. Just found one specific usecase which is causing the client to crash which I had Dm'd you. PTAL.

**Comment by AshishDhama** [comment]
> Thanks @AshishDhama. Overall the changes LGTM and is working fine. Just found one specific usecase which is causing the client to crash which I had Dm'd you. PTAL.

Thanks I don’t think crash is due to the changes I have done, But looking into it

**Review by srkgupta — APPROVED** [review]
I was able to reproduce the specified issue on latest master without this code change. Hence I have filed the issue as a separate ticket. Approving this PR. Thanks @AshishDhama

**Comment by AshishDhama** [comment]
@jgilliam17 if we delete root post from Global threads RHS (CRT), app crashes which we are able to reproduce on master,  if this issue was already known please do the needful for the following ticket @srkgupta have created for this issue [https://mattermost.atlassian.net/browse/MM-38376](https://mattermost.atlassian.net/browse/MM-38376)

**Review by jgilliam17 — APPROVED** [review]
Thank you @AshishDhama 
Tested, looks good to merge.
- Verified post is removed from the Threads view after deletion. 
Thanks for filing the crash separately @srkgupta

**Pull request merged**